src/coreclr/jit/gentree.cpp

```diff
@@ -21973,6 +21972,42 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                     break;
                 }
 
+                case TYP_LONG:
+                case TYP_ULONG:
+                {
+                    if (simdSize == 8)
+                    {
+                        // Vector64<long> vec = Vector64.CreateScalar(op1.ToScalar() * op2.ToScalar())
+                        op1            = gtNewBitCastNode(TYP_LONG, op1);
+                        op2            = gtNewBitCastNode(TYP_LONG, op2);
```

Comment:
Why bitcast instead of `ToScalar`? If this is generating better code, it seems like a pretty "core" scenario we're not handling from the `ToScalar` path

Reply:
@tannergooding because op2 can be either 8-byte `TYP_SIMD8` or 8-byte scalar so bitcast allowed me to simplify handling. In my initial version I forgot that this path is used for both `MUL(vector, vector)` and `MUL(vector, scalar)` (where `scalar` is broadcasted)

Reply:
Ah, that makes sense, 👍